### PR TITLE
Fixed audio control from media notification

### PR DESCRIPTION
**Pull request by adi23arora**
Fixes https://github.com/amahi/android/issues/357

### CODE REVIEW

File: src/main/java/org/amahi/anywhere/service/AudioService.java [hunk 1]
```diff
@@ -310,6 +314,36 @@ public void onAudioControlChange(AudioControlChangeEvent event) {
...
+
+    @Subscribe
+    public void onAudioControlPrev(AudioControlPreviousEvent event){
+        startChangedAudio(getPreviousAudioPosition());
+    }
+
+    private int getNextAudioPosition() {
+        int currentPosition = getAudioFilePosition();
+
+        if (currentPosition == audioFiles.size()-1) {
```
Comment (ArihantJain456) [inline src/main/java/org/amahi/anywhere/service/AudioService.java]: using conditional operator would be great.

**Comment by adi23arora** [comment]
@ArihantJain456  I have fixed it completely now, please review.

**Comment by ArihantJain456** [comment]
@adi23arora song is still not switching to next song on song completion (when the app is in background). You haven't changed `onCompletion` method in `AudioService`.

**Comment by adi23arora** [comment]
Ok, I'll look into it.

**Comment by ArihantJain456** [comment]
@adi23arora good work. Now it is working fine, but I have one suggestion:
AudioService and ServerFileAudioActivity both are receiving `AudioCompleteEvent` and both are trying to change song which may be unnecessary, If something can be done that prevent this then it would be great.

**Comment by adi23arora** [comment]
@ArihantJain456 We can't eliminate any one of these because of following reasons:
* In `AudioService.java`, we are using it switching audio even when the app is moved to the background.
* And in `ServerAudioFileActivity.java`, it is used for switching the view pager to update it to the current item.

**Comment by ArihantJain456** [comment]
In the `ServerAudioFileActivity.java` we are using it for both changing songs and updating view pager.
I think we should update UI (view pager in Activity) and change song in Service.
@adi23arora what do you say?

**Comment by ArihantJain456** [comment]
### The present scenario on completion of a song is as follows:
#### If the app is in background
- changing to the next song from audio service
- and play
#### If the app is not in background
- changing to the next song from service
- song changed but does not play due to some error
- updating view pager from Activity
- changing song from Activity by calling audio service
- this time song plays

### What I was suggesting
####  If the app is in background
- change song from service 
- and play
#### If the app is not in background
- update  view pager from Activity
- change song from service 
- and play

File: src/main/java/org/amahi/anywhere/service/AudioService.java [hunk 2]
```diff
@@ -316,17 +316,17 @@ public void onAudioControlChange(AudioControlChangeEvent event) {
...
     @Subscribe
     public void onAudioControlPrev(AudioControlPreviousEvent event) {
-        startChangedAudio(getPreviousAudioPosition());
+        BusProvider.getBus().post(new AudioControlChangeEvent(getPreviousAudioPosition()));
     }
 
     @Subscribe
     public void onAudioCompleted(AudioCompletedEvent event) {
-        startChangedAudio(getNextAudioPosition());
+        BusProvider.getBus().post(new AudioControlChangeEvent(getNextAudioPosition()));
```
Comment (ArihantJain456) [inline src/main/java/org/amahi/anywhere/service/AudioService.java]: This creates one additional call. So now, the audio player is preparing for the next position three times.

File: src/main/java/org/amahi/anywhere/service/AudioService.java [hunk 3]
```diff
@@ -316,17 +316,17 @@ public void onAudioControlChange(AudioControlChangeEvent event) {
 
     @Subscribe
     public void onAudioControlNext(AudioControlNextEvent event) {
-        startChangedAudio(getNextAudioPosition());
+        BusProvider.getBus().post(new AudioControlChangeEvent(getNextAudioPosition()));
```
Comment (ArihantJain456) [inline src/main/java/org/amahi/anywhere/service/AudioService.java]: Audio Player is now preparing two times.

File: src/main/java/org/amahi/anywhere/service/AudioService.java [hunk 4]
```diff
@@ -316,17 +316,17 @@ public void onAudioControlChange(AudioControlChangeEvent event) {
...
     }
 
     @Subscribe
     public void onAudioControlPrev(AudioControlPreviousEvent event) {
-        startChangedAudio(getPreviousAudioPosition());
+        BusProvider.getBus().post(new AudioControlChangeEvent(getPreviousAudioPosition()));
```
Comment (ArihantJain456) [inline src/main/java/org/amahi/anywhere/service/AudioService.java]: similar to[ line#319](https://github.com/amahi/android/pull/358/commits/98bd6cdb02e5eabde08d507f2ca68ad623f8fd2a#diff-73396d7ae4339ff7848b0e2e8d1cfdf1R319)

**Comment by ArihantJain456** [comment]
When I was testing the notification I have noticed one more issue related to the next and previous button.
Steps to reproduce:
1. click on next button in the notification
2. click on the notification to go to the audio player activity before player prepares.
3. And now the player will not play the song. <- this is the issue.

No need to rush this time.

**Comment by cpg** [comment]
Ok, sorry I had to merge the larger one and that produced conflicts. Please take a look to fix the conflicts.

**Comment by adi23arora** [comment]
@cpg I guess, work on this PR is no longer needed as the changes targeted in this were already resolved with Exoplayer implementation in PR #360.
I continued working on this because as discussed with you on IRC, I just thought exoplayer is experimental. But now when it is merged, should I close this one?

**Pull request closed without merging**